### PR TITLE
[SPARK-19659][Core]Disable spark.reducer.maxReqSizeShuffleToMem

### PR DESCRIPTION
**Pull request by zsxwing**
## What changes were proposed in this pull request?

Disable spark.reducer.maxReqSizeShuffleToMem because it breaks the old shuffle service.

Credits to @wangyum 

Closes #18466

## How was this patch tested?

Jenkins

### CODE REVIEW

**Comment by zsxwing** [comment]
cc @JoshRosen As it's impossible to safely revert #16989, I just changed the default value to Long.MaxValue.

File: docs/configuration.md [hunk 1]
```diff
@@ -529,14 +529,6 @@ Apart from these, the following properties are also available, and may be useful
   </td>
 </tr>
 <tr>
-  <td><code>spark.reducer.maxReqSizeShuffleToMem</code></td>
```
Comment (zsxwing) [inline docs/configuration.md]: don't document it as it's not a safe feature now.

File: core/src/main/scala/org/apache/spark/internal/config/package.scala [hunk 2]
```diff
@@ -326,7 +326,7 @@ package object config {
       .doc("The blocks of a shuffle request will be fetched to disk when size of the request is " +
         "above this threshold. This is to avoid a giant request takes too much memory.")
       .bytesConf(ByteUnit.BYTE)
-      .createWithDefaultString("200m")
```
Comment (JoshRosen) [inline core/src/main/scala/org/apache/spark/internal/config/package.scala]: Might want to put a `.internal()` here.

**Review by JoshRosen — APPROVED** [review]
LGTM overall. This seems like the cleanest fix for now to unblock the RC.

I do think this feature is valuable so we should see if there's a way to support it without requiring any backend changes to the shuffle service.

Let's make sure to update the "Fix Version" on the JIRA which added this feature so that it's clear that we've not shipped it in 2.2.x.

**Comment by cloud-fan** [comment]
thanks, merging to master/2.2!

**Comment by cloud-fan** [comment]
@JoshRosen I'd argue that this feature is shipped in 2.2.x, as users can change the config value to enable it. Though this is kind of an experimental feature that breaks old shuffle service.

**Comment by cloud-fan** [comment]
Oops, @wangyum sorry I forgot to set the author name... will be more careful next time :)

**Pull request closed without merging**